python/preemo/worker/_bytes.py

```diff
@@ -0,0 +1,40 @@
+from typing import Literal, TypedDict, Union, cast
+
+
+class MebibyteDict(TypedDict):
+    MiB: int
+
+
+class GibibyteDict(TypedDict):
+    GiB: int
+
+
+ByteDict = Union[MebibyteDict, GibibyteDict]
+
+KiB = 1024
+MiB = 1024 * KiB
+GiB = 1024 * MiB
+
+mebibyte_literal: Literal["MiB"] = "MiB"
+gibibyte_literal: Literal["GiB"] = "GiB"
+
+
+def convert_mebibytes_to_bytes(mebibyte_dict: MebibyteDict) -> int:
+    return mebibyte_dict[mebibyte_literal] * MiB
+
+
+def convert_gibibytes_to_bytes(gibibyte_dict: GibibyteDict) -> int:
+    return gibibyte_dict[gibibyte_literal] * GiB
+
+
+def convert_byte_dict_to_bytes(byte_dict: ByteDict) -> int:
+    # We have to cast here because mypy will not implement type narrowing for TypedDict
+    # See https://github.com/python/mypy/issues/11080
+
+    if mebibyte_literal in byte_dict:
+        return convert_mebibytes_to_bytes(cast(MebibyteDict, byte_dict))
+
+    if gibibyte_literal in byte_dict:
+        return convert_gibibytes_to_bytes(cast(GibibyteDict, byte_dict))
+
+    raise Exception(f"byte_dict had unexpected keys: {byte_dict}")
```

Comment:
if this can't have both keys anyway, does just having a `Mebibyte` and `Gibibyte` class that fulfill some interface with a `getByteSize()` make more sense

Reply:
They're both extending `TypedDict` which are kinda finicky. I can try creating a `Protocol` and having them both extend that protocol. I'm not convinced it'll work, but I'll give it a shot

Reply:
i was thinking you could drop the `typeddict`ness

Reply:
I'm not sure how to do that without changing the interface for the user.  There are certainly other user interfaces that could work just as well, but that's a separate discussion. I think to get the user interface we currently have designed, it needs to be a TypedDict